lib/ansible/modules/files/find.py

```diff
@@ -33,7 +33,7 @@
         default: '*'
         description:
             - One or more (shell or regex) patterns, which type is controlled by C(use_regex) option.
-            - The patterns restrict the list of files to be returned to those whose basenames match at
+            - The patterns restrict the list of files to be returned to those whose basenames or complete filenames (since version 2.5) match at
```

Comment:
thinking of this further, it will break existing playbooks that rely on basename only, so this really requires a toggle 'match_full_name' or similar and the default behaviour should continue to be what it is now.